CMSgov/mint-app
# Noref/operational solutions refactor be

## Changes and Description

-Operational Solutions is returned as one array instead of separating Possible and actual
   - archived changed to needed. Default value is true
   - Collection of solutions on OperationalNeed has an optional parameter of includedUnneeded
   - Same query available by operational Need
-Single Query to get a specific Operational Need by ID
- Single Query to get a specific Operational Solution by ID

-Unit tests updated

## How to test this change
1. Import the updated Postman Collection.
2. Create a Model Plan
3. Update the OperationalNeedId Variable to a relevant one in the collection ( Or create a custom need)
4. Query the unneeded solutions to ensure that they are there (EG all possible solutions for that Need)
5. Insert a solution
6. Query to get the solution.

## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
